sound/soc/sof/amd/acp-ipc.c

```diff
@@ -165,7 +173,7 @@ irqreturn_t acp_sof_ipc_irq_thread(int irq, void *context)
 	if (dsp_msg) {
 		snd_sof_ipc_msgs_rx(sdev);
 		acp_dsp_ipc_host_done(sdev);
-		ipc_irq = true;
+		return IRQ_HANDLED;
```

Comment:
this changes the flow. With the previous code, you would have tried to deal with the dsp_ack. That sounds like a different change that needs to go in a different patch, if that was intentional indeed.

Reply:
Here we are handling one to one communication from DSP to HOST in synchronous manner. we never get two conditions met, so we optimized and done.

Reply:
separate change then. this has nothing to do with the panic stuff

Reply:
Changes Done ..Optimization patch will be done later.

Only added code related to DSP panic Stuff.